Labs/Lab Notebooks/05 - Azure Cognitive Search/embed_pdf_edits.ipynb

```diff
@@ -0,0 +1,399 @@
+    "        prioritized_keywords_fields=[SemanticField(field_name=\"category\")],\n",
+    "        prioritized_content_fields=[SemanticField(field_name=\"content\")]\n",
+    "    )\n",
+    ")\n",
+    "\n",
+    "# Create the semantic settings with the configuration\n",
+    "semantic_settings = SemanticSettings(configurations=[semantic_config])\n",
+    "\n",
+    "# Create the search index with the semantic settings\n",
+    "index = SearchIndex(name=index_name, fields=fields,\n",
+    "                    vector_search=vector_search, semantic_settings=semantic_settings)\n",
+    "result = index_client.create_or_update_index(index)\n",
+    "print(f' {result.name} created')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "### function for breaking up document into different chunks\n",
+    "\n",
+    "def split_text(text, chunk_size: int, chunk_overlap: int = 0, length_function: Callable[[str], int] = len):\n",
+    "    text_splitter = RecursiveCharacterTextSplitter(\n",
+    "        chunk_size = chunk_size,\n",
+    "        chunk_overlap = chunk_overlap,\n",
+    "        length_function = length_function\n",
+    "    )\n",
+    "    split_text = text_splitter.create_documents([text])\n",
+    "    \n",
+    "    return split_text"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 24,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Found DTO FAQ.pdf\n",
+      "Uploaded 31 chunks for DTO FAQ.pdf\n",
+      "Found DTO Overview.pdf\n",
+      "Uploaded 13 chunks for DTO Overview.pdf\n",
+      "Found DTO Quick Guide.pdf\n",
+      "Uploaded 4 chunks for DTO Quick Guide.pdf\n",
+      "Found Vacation and Holidays - Hourly_Non-exempt Corporate Employees.pdf\n",
+      "Uploaded 14 chunks for Vacation and Holidays - Hourly_Non-exempt Corporate Employees.pdf\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Embed and Upload Blobs to Azure Cognitive Search\n",
+    "import tempfile\n",
+    "import uuid\n",
+    "\n",
+    "blob_client = BlobServiceClient(\n",
+    "   account_url=storage_url,\n",
+    "   credential=storage_key\n",
+    ")\n",
+    "\n",
+    "raw_files_container = blob_client.get_container_client(storage_container_name)\n",
+    "\n",
+    "filenames = raw_files_container.list_blob_names()\n",
+    "\n",
+    "@retry(wait=wait_random_exponential(min=1, max=20), stop=stop_after_attempt(6))\n",
+    "# Function to generate embeddings for title and content fields, also used for query embeddings\n",
+    "def generate_embeddings(text):\n",
+    "    response = openai.embeddings.create(\n",
+    "        input=text,model=\"text-embedding-ada-002\"\n",
+    "    )\n",
+    "    embeddings = response.data[0].embedding\n",
+    "    return embeddings\n",
+    "\n",
+    "\n",
+    "chunk_size = 1000\n",
```

Comment:
is there a reason why these chunking numbers were chosen? I was under assumption a bit larger with some overlap would be more ideal

Reply:
No particular reason. Just wanted to test the entire setup out.